common/src/main/java/org/keycloak/common/crypto/CryptoIntegration.java

```diff
@@ -0,0 +1,58 @@
+package org.keycloak.common.crypto;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.ServiceLoader;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+import org.jboss.logging.Logger;
+
+/**
+ * @author <a href="mailto:mposolda@redhat.com">Marek Posolda</a>
+ */
+public class CryptoIntegration {
+
+    protected static final Logger logger = Logger.getLogger(CryptoIntegration.class);
+
+    private static volatile CryptoProvider securityProvider;
+
+    private static volatile ClassLoader classLoader;
+
+    public static void setClassLoader(ClassLoader classLoader) {
+        CryptoIntegration.classLoader = classLoader;
+    }
+
+    public static CryptoProvider getProvider() {
+        if (securityProvider == null) {
+            securityProvider = detectProvider();
+            logger.infof("Detected security provider: %s", securityProvider.getClass().getName());
+        }
+        return securityProvider;
+    }
```

Comment:
This isn't concurrency safe? Could it be better to instead rename `setClassLoader` to `init` and have it create instantiate `securityProvider`.

Reply:
+1, will update that